src/server/session.ts

```diff
@@ -26,14 +26,7 @@ namespace ts.server {
     }
 
     function shouldSkipSematicCheck(project: Project) {
-        if (project.getCompilerOptions().skipLibCheck !== undefined) {
-            return false;
-        }
-
-        if ((project.projectKind === ProjectKind.Inferred || project.projectKind === ProjectKind.External) && project.isJsOnlyProject()) {
-            return true;
-        }
-        return false;
+        return (project.projectKind === ProjectKind.Inferred || project.projectKind === ProjectKind.External) && project.isJsOnlyProject();

```

Comment:
So if they're using a "configured" project (e.g. a lot of our solutions for issues such as "too much JavaScript" involve adding a tsconfig with `"allowJs": true` and settings some exclude dirs), will they get semantic diagnostics again? If its a configured project with "allowJs" set and it `isJsOnlyProject`, then it should also suppress semantic diagnostics.